tests/test_update_modules.py

```diff
@@ -0,0 +1,76 @@
+#!/usr/bin/python
+# Copyright 2021 Northern.tech AS
+#
+#    Licensed under the Apache License, Version 2.0 (the "License");
+#    you may not use this file except in compliance with the License.
+#    You may obtain a copy of the License at
+#
+#        http://www.apache.org/licenses/LICENSE-2.0
+#
+#    Unless required by applicable law or agreed to in writing, software
+#    distributed under the License is distributed on an "AS IS" BASIS,
+#    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#    See the License for the specific language governing permissions and
+#    limitations under the License.
+
+
+import os
+import shutil
+import subprocess
+import tempfile
+
+import pytest
+
+from utils.common import put_no_sftp
+
+
+@pytest.mark.usefixtures("setup_board", "bitbake_path")
+class TestUpdateModules:
+    @pytest.mark.min_mender_version("2.0.0")
+    def test_directory_update_module(self, connection):
+        """Test the directory based update module, first with a failed update,
+        then a successful one."""
+
+        file_tree = tempfile.mkdtemp()
+        try:
+            files_and_content = ["file1", "file2"]
+            for file_and_content in files_and_content:
+                with open(os.path.join(file_tree, file_and_content), "w") as fd:
+                    fd.write(file_and_content)
+
+            artifact_file = os.path.join(file_tree, "update.mender")
+            cmd = (
+                "directory-artifact-gen -o %s -n update-directory -t qemux86-64 -d /tmp/test_directory_update_module %s"
```

Comment:
`qemux86-64` needs to be replaced with `bitbake_variables["MACHINE"]`, and you need to include `bitbake_variables` in the test arguments to import the fixture. I would expect this to fail on vexpress-qemu without it.

There is one more below, make sure you get both.